src/contractor/contractor.cpp

```diff
@@ -495,11 +495,13 @@ std::size_t Contractor::LoadEdgeExpandedGraph(
 
         // vector to count used speeds for logging
         // size offset by one since index 0 is used for speeds not from external file
-        std::vector<std::atomic<std::uint64_t>> segment_speeds_counters(
-            segment_speed_filenames.size() + 1);
-        for (auto &each : segment_speeds_counters)
-            each.store(0);
-        const constexpr auto LUA_SOURCE = 0;
+        std::vector<std::atomic<std::uint64_t>> segment_speeds_counters;
+        for (std::size_t i=0;i<segment_speed_filenames.size() + 1;++i)
+       {
+            segment_speeds_counters.emplace_back();
+            segment_speeds_counters[i].store(0);
+        }
```

Comment:
@daniel-j-h is it really a bug in the standard library or clang-3.5 itself? I'm asking because the standard library was the same on the system this replicates on and g++ was not impacted, only clang.